test-data/unit/fine-grained.test

```diff
@@ -1809,8 +1809,8 @@ def f() -> Iterator[None]:
 [typing fixtures/typing-medium.pyi]
 [builtins fixtures/list.pyi]
 [triggered]
-2: <b>, __main__
-3: <b>, __main__, a
+2: <b>, <b[wildcard]>, __main__
+3: <b>, <b[wildcard]>, __main__, a
```

Comment:
So.... the addition of `dict` in fixtures/list.pyi now causes `__annotations__` to exist:
https://github.com/python/mypy/blob/bac9e77eaeb36c0535bc05cb1faf6eced25b8af1/mypy/semanal.py#L637-L639

which then makes it one of the names that's part of the diff:
https://github.com/python/mypy/blob/a08388cf6de053f659a9d663387f8f71e68664d8/mypy/server/update.py#L781-L795
(sounds reasonable to me)

Generally I noticed 3 approaches when a built-in is missing:
- crash — like when handling **kwargs here:
   https://github.com/python/mypy/blob/e778a58066f23982d5cbe1df5317d6540c5902fd/mypy/types.py#L3607
- avoid doing a thing — like when not adding `__annotations__` above
- fallback to `object` — like when deciding how to type `NamedTuple._field_types`:
   https://github.com/python/mypy/blob/dcf910e05fdcff00b1fb9a8b9ca85f9a6667b946/mypy/semanal_namedtuple.py#L484-L486

Granted it's a "testing problem" since otherwise those builtins can be taken for granted, but would be good to have a shared/recommended approach to make tests maintainable (would rather not have to do whack-a-mole in unrelated tests)... cc @97littleleaf11 @hauntsaninja